gi_core/dispatch/math/vectorized/gamma_index_range_impl.cpp

```diff
@@ -0,0 +1,594 @@
+#include <math/vectorized/gamma_index_range_impl.hpp>
+
+#include <simdpp/dispatch/get_arch_raw_cpuid.h>
+#define SIMDPP_USER_ARCH_INFO ::simdpp::get_arch_raw_cpuid()
+
+namespace yagit::core::math::vectorized::detail
+{
+	namespace SIMDPP_ARCH_NAMESPACE
+	{
+		// -------- float32 --------
+
+		constexpr size_t float32_VectorSize = SIMDPP_FAST_FLOAT32_SIZE;
```

Comment:
Add a comment explaining WHY this code is so much duplicated:
 - dimensions: 1,2,3
 - true/false alignment
 - float32 vs double

Explain why this cannot be done on templates  or in any shorter form.